src/Features/CSharp/Portable/UseAnotherLiteral/CSharpUseAnotherLiteralCodeFixProvider.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.CSharp.UseAnotherLiteral
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp, Name = PredefinedCodeFixProviderNames.UseAnotherLiteral), Shared]
+    internal sealed class CSharpUseAnotherLiteralCodeFixProvider : CodeFixProvider
+    {
+        private const string CS8313 = nameof(CS8313); // A default literal 'default' is not valid as a case constant. Use another literal (e.g. '0' or 'null') as appropriate. If you intended to write the default label, use 'default:' without 'case'.
+        private const string CS8363 = nameof(CS8363); // A default literal 'default' is not valid as a pattern. Use another literal (e.g. '0' or 'null') as appropriate. To match everything, use a discard pattern 'var _'.
+
+        public override ImmutableArray<string> FixableDiagnosticIds { get; } =
+            ImmutableArray.Create(CS8313, CS8363);
+
+        public override FixAllProvider GetFixAllProvider()
+        {
+            // This code fix addresses very specific compiler errors. It's unlikely there will be more than 1 of them at a time.
+            return null;
+        }
+
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var syntaxRoot = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+            var token = syntaxRoot.FindToken(context.Span.Start);
+
+            if (token.Span == context.Span &&
+                token.IsKind(SyntaxKind.DefaultKeyword) &&
+                token.Parent.IsKind(SyntaxKind.DefaultLiteralExpression))
+            {
+                var defaultLiteral = (LiteralExpressionSyntax)token.Parent;
+                var semanticModel = await context.Document.GetSemanticModelAsync(context.CancellationToken).ConfigureAwait(false);
+
+                var constant = semanticModel.GetConstantValue(defaultLiteral, context.CancellationToken);
+                if (!constant.HasValue)
+                {
+                    return;
+                }
+
+                var newLiteral = SyntaxGenerator.GetGenerator(context.Document).LiteralExpression(constant.Value);
```

Comment:
nit.  this could be delayed (though i don't care much) :)

Reply:
It's here so that we can put the literal in the code fix title. Would you prefer a generic title instead?

Reply:
ah ok.  makes sense then :)